### ISSUE TITLE
Secrets not working with `bash .command.run`

### ISSUE THREAD

**Issue by nick-youngblut** [nextflow-io/nextflow#5613]
### Expected behavior and actual behavior

AFAIK, to debug a process, one normally moves to the process working directory and runs `bash .command.run`.

At least with Nextflow 24.10.2, this results in `grep: /home/nickyoungblut/.nextflow/secrets/.nf-0f268357-4937-450a-a7ec-9f343ae363e9.secrets: No such file or directory`, and the secret is not provided for the process. 

I believe that for previous versions of Nextflow, the secret was provided to the process script when directly calling `bash .command.run`.

**Also, in regards to the docs:**

> This feature is only available when using the local or grid executors (Slurm, Grid Engine, etc). The AWS Batch executor allows the use of secrets when deploying the pipeline execution via [Seqera Platform](https://seqera.io/blog/pipeline-secrets-secure-handling-of-sensitive-information-in-tower/?__hstc=247481240.55878b6ead92ce71b87483addc8adb16.1715875238933.1733965286501.1734293347382.101&__hssc=247481240.1.1734293347382&__hsfp=4002305717).

There is no mention of GCP. Must one also use the [Seqera Platform](https://seqera.io/blog/pipeline-secrets-secure-handling-of-sensitive-information-in-tower/?__hstc=247481240.55878b6ead92ce71b87483addc8adb16.1715875238933.1733965286501.1734293347382.101&__hssc=247481240.1.1734293347382&__hsfp=4002305717) to utilize secrets if using GCP (instead of AWS)? a

### Steps to reproduce the problem

Set a secret via `nextlfow secret set my_secret my_secret_value` and use it in a process via `secret "my_secret"`. When the process is executed via `nextflow run`, the secret is provided. However, the secret is not provided if running `bash .command.run`. 

### Environment 

* Nextflow version: 24.10.2
* Java version: 11.0.1
* Operating system: 22.04.4
* Bash version: 5.1.16

**Comment by nick-youngblut** [comment]
According to the `.command.run` file:

```
source /dev/stdin <<<"$(cat <(grep -w -e 'my_secret=.*' /home/nickyoungblut/.nextflow/secrets/.nf-4aa05d79-d6d0-464b-a166-b01307897573.secrets))"
```

`/home/nickyoungblut/.nextflow/secrets/` only contains `store.json`.

The `.command.run` file is incorrectly assuming that `*.secrets` exists.

**Comment by nick-youngblut** [comment]
Is `.nextflow/secrets/.nf-*` deleted right after the job run finishes? If so, how could one actually debug any job that includes a secret, as noted in the standard Nextflow job failure message:

> Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`

The `.command.sh` script does not create the `.nextflow/secrets/.nf-*`. It just looks for the existing file via:

```
source /dev/stdin <<<"$(cat <(grep -w -e 'my_secret=.*' /home/nickyoungblut/.nextflow/secrets/.nf-4aa05d79-d6d0-464b-a166-b01307897573.secrets))"
```

**Comment by pditommaso** [comment]
You are right, this could be a drawback of this change https://github.com/nextflow-io/nextflow/commit/e28537ca. We need to revert it

**Comment by pditommaso** [comment]
Solved by [#5697](https://github.com/nextflow-io/nextflow/pull/5697)
